src/diffusers/models/unet_2d_blocks.py

```diff
@@ -317,6 +345,35 @@ def get_up_block(
             attn_num_head_channels=attn_num_head_channels,
             resnet_time_scale_shift=resnet_time_scale_shift,
         )
+    elif up_block_type == "KUpBlock2D":
+        return KUpBlock2D(
+            num_layers=num_layers,
+            in_channels=in_channels,
+            out_channels=out_channels,
+            temb_channels=temb_channels,
+            add_upsample=add_upsample,
+            resnet_eps=resnet_eps,
+            resnet_act_fn=resnet_act_fn,
+        )
+    elif up_block_type == "KCrossAttnUpBlock2D":
+        if cross_attention_dim is None:
+            raise ValueError("cross_attention_dim must be specified for KCrossAttnUpBlock2D")
+        is_first_block = in_channels == out_channels == temb_channels
```

Comment:
Can we move this logic maybe directly into the block so that this "get" function stays clean?

Reply:
✅